src/framework/json.hpp

```diff
@@ -256,6 +313,173 @@ void std::to_json(json_t &js, const std::map<uint64_t, T1, T2> &map) {
   }
 }
 
+template <typename T>
+json_t numpy_to_json_1d(py::array_t<T, py::array::c_style> arr) {
+    py::buffer_info buf = arr.request();
+    if (buf.ndim != 1) {
+        throw std::runtime_error("Number of dims must be 1");
+    }
+
+    T *ptr = (T *) buf.ptr;
+    size_t D0 = buf.shape[0];
+
+    //std::cout << "1-d conversion: " << X << std::endl;
+
+    std::vector<T> tbr;
+    for (size_t n0 = 0; n0 < D0; n0++)
+        tbr.push_back(ptr[n0]);
```

Comment:
hmm wondering if `py::array_t<T, py::array::c_style>` can be used with `std::copy()` so we just:
```c++
std::copy(arr.begin(), arr.end(), std::back_inserter(tbr));
```
Did you try it?
If this work, this method is a one liner
... and we could get rid of all the other dimensional `numpy_to_json` methods

Reply:
So this one is a bit nuanced - for the 1-d case, yes, it's quite simple and there are for sure more efficient ways - for example just doing `return std::vector<T>(arr, arr + D0);` would avoid the dynamic resizing caused by initializing an empty vector and then appending (which both the current and your suggested solution suffer from).

I'll take another look at these and see if I can't find a better solution because I think I know what you're getting at. 

As an aside, the reason I was semi-okay with the way they are now is that it clearly shows how they could be extended to 4,5,...-d tensors by someone who has no previous knowledge. The three functions follow a similar pattern with the most significant difference being the indexing function in the middle of the nested loops: 
1-d: `n0`
2-d: `n1+D1*n0`
3-d: `n2 + D2*(n1 + D1*n0)`
So it basically shows the [c-style array indexing equation](https://eli.thegreenplace.net/2015/memory-layout-of-multi-dimensional-arrays/), which I think would make it easier for whoever wanted to add 4-d, 5-d, etc. 

Of course, it would be better if it just handled any dimension, perfectly efficiently, right now! Which is what I will look into.